src/cgame/cg_event.cpp

```diff
@@ -617,19 +617,7 @@ void CG_OnPlayerWeaponChange()
 	// Change the HUD to match the weapon. Close the old hud first
 	Rocket_ShowHud( ps->weapon );
 
-	// Rebuild weapon lists if UI is in focus.
-	if ( trap_Key_GetCatcher() == KEYCATCH_UI && ps->persistant[ PERS_TEAM ] == TEAM_HUMANS )
-	{
-		CG_Rocket_BuildArmourySellList( "default" );
-		CG_Rocket_BuildArmouryBuyList( "default" );
-
-		Rocket::Core::ElementDocument* document = menuContext->GetDocument( rocketInfo.menu[ ROCKETMENU_ARMOURYBUY ].id );
-		if ( document->IsVisible() )
-		{
-			document->DispatchEvent( "refreshdata", {} );
-		}
-	}
-
+	CG_OnPlayerUpgradeChange();
```

Comment:
Rename this to "rebuild armoury menu" or something like that

Reply:
It's just reusing existing code I thing, I don't think I came with such a name myself? Or do you mean renaming "CG_OnPlayerWeaponChange" and not "CG_OnPlayerUpgradeChange" ? I thought you'd prefer minimal changes.

Reply:
Rename CG_OnPlayerUpgradeChange because it's going to be used for things other than upgrade changes.